src/ai_channel.rs

```diff
@@ -108,9 +109,21 @@ pub async fn serve(
             break;
         }
 
-        let system_prompt = ChatCompletionRequestMessage::System(
-            include_str!("./ai_channel/system_prompt.txt").into(),
-        );
+        let current_prompt = {
+            let Ok(data) = shared_prompt.lock() else {
+                tracing::error!("Shared prompt invalid; Will re-check prompt after a wait.");
+                tracing::error!(
+                    "If this doesn't fix itself edit the system_prompt file to force a reload."
+                );
+
+                tokio::time::sleep(Duration::from_secs(5)).await;
+
+                continue;
+            };
+
+            //TODO(tye): does this clone the data everytime. If so, is that too expensive?
```

Comment:
Not a suggestion but quick fyi: it does, but it's totally fine here. There is no other way around it using the library we use. The actual API request will take much longer anyway!